volatility/framework/plugins/windows/getsids.py

```diff
@@ -0,0 +1,257 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+# Creator: Aviel Zohar (memoryforensicsanalysis@gmail.com) # Fix vol2 plugin to vol3
+import logging
+from typing import Callable, List, Generator, Iterable, Dict
+import re, ntpath
+
+from volatility.framework import renderers, interfaces, objects, exceptions
+from volatility.framework.configuration import requirements
+from volatility.framework.objects import utility
+from volatility.framework.renderers import format_hints
+from volatility.plugins.windows import pslist
+from volatility.framework.layers.registry import RegistryFormatException
+from volatility.framework.symbols.windows.extensions.registry import RegValueTypes
+import volatility.plugins.windows.getservicesids as getservicesids
+import volatility.plugins.windows.registry.hivelist as hivelist
+
+
+vollog = logging.getLogger(__name__)
+
+
+
+def find_sid_re(sid_string, sid_re_list) -> str:
+    for reg, name in sid_re_list:
+        if reg.search(sid_string):
+            return name
+
+well_known_sid_re = [
+  (re.compile(r'S-1-5-[0-9-]+-500$'), 'Administrator'),
+  (re.compile(r'S-1-5-[0-9-]+-501$'), 'Guest'),
+  (re.compile(r'S-1-5-[0-9-]+-502$'), 'KRBTGT'),
+  (re.compile(r'S-1-5-[0-9-]+-512$'), 'Domain Admins'),
+  (re.compile(r'S-1-5-[0-9-]+-513$'), 'Domain Users'),
+  (re.compile(r'S-1-5-[0-9-]+-514$'), 'Domain Guests'),
+  (re.compile(r'S-1-5-[0-9-]+-515$'), 'Domain Computers'),
+  (re.compile(r'S-1-5-[0-9-]+-516$'), 'Domain Controllers'),
+  (re.compile(r'S-1-5-[0-9-]+-517$'), 'Cert Publishers'),
+  (re.compile(r'S-1-5-[0-9-]+-520$'), 'Group Policy Creator Owners'),
+  (re.compile(r'S-1-5-[0-9-]+-533$'), 'RAS and IAS Servers'),
+  (re.compile(r'S-1-5-5-[0-9]+-[0-9]+'), 'Logon Session'),
+  (re.compile(r'S-1-5-21-[0-9-]+-518$'), 'Schema Admins'),
+  (re.compile(r'S-1-5-21-[0-9-]+-519$'), 'Enterprise Admins'),
+  (re.compile(r'S-1-5-21-[0-9-]+-553$'), 'RAS Servers'),
+  (re.compile(r'S-1-5-21-[0-9-]+-498$'), 'Enterprise Read-Only Domain Controllers'),
+  (re.compile(r'S-1-5-21-[0-9-]+-521$'), 'Read-Only Domain Controllers'),
+  (re.compile(r'S-1-5-21-[0-9-]+-522$'), 'Cloneable Domain Controllers'),
+  (re.compile(r'S-1-5-21-[0-9-]+-525$'), 'Protected Users'),
+  (re.compile(r'S-1-5-21-[0-9-]+-553$'), 'Remote Access Services (RAS)'),
+]
+
+well_known_sids = {

```

Comment:
This also seems like data and would be better placed in a JSON file that can be loaded/modified without changing code.